src/unix/notbsd/linux/other/mod.rs

```diff
@@ -1,5 +1,110 @@
 pub type __priority_which_t = ::c_uint;
 
+cfg_if! {
+    if #[cfg(libc_union)] {
+        #[repr(C)]
+        #[derive(Copy, Clone)]
+        struct addr_bnd_t {
+            lower: *mut ::c_void,
+            upper: *mut ::c_void,
+        }
+
+        #[repr(C)]
+        #[derive(Copy, Clone)]
+        union sigfault_t_anonymous_union {
+            addr_bnd: addr_bnd_t,
+            pkey: u32,
+        }
+
+        #[repr(C)]
+        #[derive(Copy, Clone)]
+        struct sigfault_t {
+            addr: *mut ::c_void,
+            #[cfg(target_arch = "sparc")]
+            trapno: ::c_int,
+            addr_lsb: ::c_short,
+            anonymous_union: sigfault_t_anonymous_union,
+        }
+
+        #[repr(C, packed)]
+        #[derive(Copy, Clone)]
+        union sifields_t {
+            _pad: [::c_int; 29],
```

Comment:
So this is interesting. There's a disagreement between `/usr/include/asm-generic/siginfo.h` and `/usr/include/x86_64-linux-gnu/bits/siginfo.h` on my system about whether `_pad` is 28 or 29 elements. They produce the same overall struct size because of padding/alignment, but I'm not sure what the right value to put here is?

Reply:
Hm, this is weird. If the struct size is the same, I think I'd rather pick 29, but... this might be a bug that you might want to report upstream.